pykafka/cluster.py

```diff
@@ -287,4 +288,8 @@ def update(self):
                         'will NOT work. You need to create at least one topic '
                         'manually using the Kafka CLI tools.')
         self._update_brokers(metadata.brokers)
-        self._update_topics(metadata.topics)
+        try:
+            self._update_topics(metadata.topics)
+        except LeaderNotAvailable:
```

Comment:
If we're going to discard the exception here, there might be a case for just discarding it directly in `Partition.update`, and logging it from there: that way, an exception on one Partition wouldn't stop us from updating remaining Topics/Partitions, and the leaderless Partition may not be one that this client cares about.